File: sdks/dotnet/src/Mobiscroll.Connect/Exceptions/RateLimitException.cs
namespace Mobiscroll.Connect.Exceptions;

public sealed class RateLimitException : MobiscrollConnectException
{
    /// <summary>Value from the Retry-After header, in seconds, if present.</summary>
    public int? RetryAfter { get; }

    public RateLimitException(string message, int? retryAfter = null)
        : base(message, "RATE_LIMIT_ERROR")
    {
        RetryAfter = retryAfter;
    }
}

File: sdks/java/connect-sdk/src/main/java/com/mobiscroll/connect/exceptions/RateLimitException.java
package com.mobiscroll.connect.exceptions;

/** Thrown on HTTP 429. */
public class RateLimitException extends MobiscrollConnectException {

    private static final long serialVersionUID = 1L;

    private final Integer retryAfter;

    public RateLimitException(String message, Integer retryAfter) {
        super(message);
        this.retryAfter = retryAfter;
    }

    /** Seconds parsed from the {@code Retry-After} response header, or {@code null} if absent or unparsable. */
    public Integer getRetryAfter() {
        return retryAfter;
    }
}
